### ISSUE TITLE
ParameterException is not thrown when a parameter with variable arity has no value

### ISSUE THREAD

**Issue by 7acd22f0bcd3ac948246** [cbeust/jcommander#343]
**Description**

ParameterException is not thrown when a parameter with variable arity has no value.
This only happens if the parameter having no value is located inside the list of parameters.
If the parameter is located at the end of the parameter list and no value is provided for it, then a proper  ParameterException is thrown.

**Expected result:**
ParameterException is thrown.

**Actual result:**
No ParameterException is thrown 

**Environment:** 
 - Java version: 1.8.0_121
- JCommander version: 1.66

**Steps to reproduce:**

Have the following parameter setup:

```java
import com.beust.jcommander.Parameter;

import java.util.List;

public class TestParams {
    @Parameter(
            names = {"-first"},
            description = "first parameter",
            required = true,
            validateWith = First.class)
    private String first;

    @Parameter(
            names = {"-second"},
            description = "second parameter",
            variableArity = true,
            validateWith = Second.class)
    private List<String> second;

    @Parameter(
            names = {"-third"},
            description = "third parameter",
            variableArity = true,
            validateWith = Third.class)
    private List<String> third;
}
```

The following call should throw a ParameterException since the parameter **-second** has no value:
```shell
<main class> -first value1 -second -third value3
```

If the parameter having variable arity but no value, is located at the end of the list, a proper ParameterException is thrown.
I believe this is the correct behavior.
The following call throws a ParameterException since the parameter **-third** has no value:

```shell
<main class> -first value1 -second value2 -third
```

**Additional suggestion**

Maybe there should be a way to also specify the minimum expected number of values for parameters with variable arity.

**Comment by cbeust** [comment]
Thanks, will take a look.

As for your second question, `variableArity` transfers full control of the parsing to the user's code, so I don't think it's necessary to specify a minimum number of parameters in an annotation: the code will check that.

**Comment by 7acd22f0bcd3ac948246** [comment]
Thank you.

I did some more checking and it seems that if I implement the IVariableArity interface and execute the call as described in the above issue, the parameter **-third** ends up as a value for the parameter **-second**.

For the parameter setup described below consider the following call:
```shell
<main class> -first value1 -second -third value3
```
```java
import com.beust.jcommander.IVariableArity;
import com.beust.jcommander.Parameter;

import java.util.List;

public class TestParams implements IVariableArity {
    @Parameter(
            names = {"-first"},
            description = "first parameter",
            required = true,
            validateWith = First.class)
    private String first;

    @Parameter(
            names = {"-second"},
            description = "second parameter",
            variableArity = true,
            validateWith = Second.class)
    private List<String> second;

    @Parameter(
            names = {"-third"},
            description = "third parameter",
            variableArity = true,
            validateWith = Third.class)
    private List<String> third;

    @Override
    public int processVariableArity(String s, String[] strings) {
        System.out.println("parameter: " + s);  

        for (String value : strings) {
            System.out.println("value: " + value);
        }

        return strings.length;
    }
}
```

The parameter **-third** ends up as a value for the parameter **-second**:

```shell
s = "-second"
strings = {"-third", "value3"}
```

**Comment by 7acd22f0bcd3ac948246** [comment]
@cbeust Did you, by any chance, had some time to take a look at this? 
Thank you

**Comment by cbeust** [comment]
Not yet.


-- 
Cédric

**Comment by 7acd22f0bcd3ac948246** [comment]
@cbeust 
The issue seems to be that, in this particular case, for the parameter **-second**, determineArity() will return an arity of '**0**' ([line 823 in JCommander.java](https://github.com/cbeust/jcommander/blob/master/src/main/java/com/beust/jcommander/JCommander.java#L823-L825)) and currently, this case is not handled for non-boolean parameters in [processFixedArity()](https://github.com/cbeust/jcommander/blob/master/src/main/java/com/beust/jcommander/JCommander.java#L838), where a ParameterException should be thrown.
If I understand correctly, an arity of '**0**' should only be allowed for boolean parameters.

**Comment by cbeust** [comment]
You're welcome to submit a PR if you think you have a fix! (along with a test if possible)

Thanks!

**Comment by 7acd22f0bcd3ac948246** [comment]
Fixed by the merge of [PR #348](https://github.com/cbeust/jcommander/commit/fadaa32e3113690dc0cdd0fe66cffeddaced9daf)
